src/travel-details-screens/TripDetailsScreenComponent.tsx

```diff
@@ -151,73 +151,155 @@ export const TripDetailsScreenComponent = ({
-    useGetFirstTariffZoneWeSellTicketFor(toTariffZones);
+  const isFromTravelSearchToTicketEnabled =
+    useFromTravelSearchToTicketEnabled();
+  const {fareProductTypeConfigs} = useFirestoreConfiguration();
+  const {tariffZones} = useFirestoreConfiguration();
+  const {data: harbors} = useHarbors();
 
   const hasTooLongWaitTime = totalWaitTimeIsMoreThanAnHour(tripPattern.legs);
-  const canSellCollab = canSellCollabTicket(tripPattern);
 
   if (
-    !(
-      fromTripsSearchToTicketEnabled &&
-      fromTariffZoneWeSellSingleTicketsFor &&
-      toTariffZoneWeSellTicketFor
-    ) ||
+    !enable_ticketing ||
+    !isFromTravelSearchToTicketEnabled ||
     hasTooLongWaitTime
   )
     return;
 
-  const tariffZoneTo: TariffZoneWithMetadata = {
-    resultType: 'zone',
-    venueName: nonHumanLegs[nonHumanLegs.length - 1]?.toPlace?.name,
-    ...toTariffZoneWeSellTicketFor,
-  };
-  const tariffZoneFrom: TariffZoneWithMetadata = {
-    resultType: 'zone',
-    venueName: nonHumanLegs[0]?.fromPlace?.name,
-    ...fromTariffZoneWeSellSingleTicketsFor,
-  };
+  const nonHumanLegs = getNonHumanLegs(tripPattern.legs);
+  const ticketStartTime = calculateTicketStartTime(nonHumanLegs);
+
+  const ticketInfoForBus = getTicketInfoForBus(
+    tripPattern,
+    nonHumanLegs,
+    fareProductTypeConfigs,
+    tariffZones,
+    ticketStartTime,
+  );
+  if (ticketInfoForBus) return ticketInfoForBus;
+
+  const ticketInfoForBoat = getTicketInfoForBoat(
+    tripPattern,
+    nonHumanLegs,
+    fareProductTypeConfigs,
+    harbors,
+    ticketStartTime,
+  );
+  if (ticketInfoForBoat) return ticketInfoForBoat;
+}
+
+function getNonHumanLegs(legs: Leg[]) {
+  return legs.filter(
+    (leg) => leg.mode !== Mode.Foot && leg.mode !== Mode.Bicycle,
+  );
+}
+
+function calculateTicketStartTime(legs: Leg[]) {
+  const tripStartWithBuffer = addMinutes(parseISO(legs[0]?.aimedStartTime), -5);
+  return tripStartWithBuffer.getTime() <= Date.now()
+    ? undefined
+    : formatISO(tripStartWithBuffer);
+}
 
-  // modes we can sell single tickets for. Might not always match modes we sell tickets for,
-  // as from travel search to ticket currently only supports single ticket
-  const someLegsAreNotSingleTicket = hasLegsWeCantSellTicketsFor(tripPattern, [
+function getTicketInfoForBus(
+  tripPattern: TripPattern,
+  nonHumanLegs: Leg[],
+  fareProductTypeConfigs: FareProductTypeConfig[],
+  tariffZones: TariffZone[],
+  ticketStartTime?: string,
+) {
+  const canSellCollab = canSellCollabTicket(tripPattern);
+  const hasOnlyValidBusLegs = !hasLegsWeCantSellTicketsFor(tripPattern, [
     'cityTram',
     'expressBus',
     'localBus',
     'localTram',
     'regionalBus',
     'shuttleBus',
   ]);
-  if (!enable_ticketing || (someLegsAreNotSingleTicket && !canSellCollab))
-    return;
 
-  const tripStartWithBuffer = addMinutes(
-    parseISO(nonHumanLegs[0]?.aimedStartTime),
-    -5,
+  if (!hasOnlyValidBusLegs && !canSellCollab) return;
```

Comment:
Not 100% sure what `canSellCollab` is, but it will never be `true` if `hasOnlyValidBusLegs` is `false`?

Reply:
Not that it matters in this case, since the condition looks to be correct either way. 👌

Reply:
Or... wait. Original question is still valid.

Reply:
The `canSellCollab` check is AtB-specific code to account for single ticket bus being valid on train trips in zone A. `canSellCollab` is true if some legs in the trip are by train in zone A (only for AtB). 

It can be `true` if `hasOnlyValidBusLegs` is `false` because still some legs might be by train in zone A, but the trip can also include a boat journey for example